docs/reference/aggregations.asciidoc

```diff
@@ -3,117 +3,367 @@
+aggregations return different aggregations types depending on the data type of
+the aggregated field.
+
+[source,console-result]
+----
+{
+  ...
+  "aggregations": {
+    "histogram#my-agg-name": {                 <1>
+      "buckets": []
+    }
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took": "$body.took", "timed_out": false, "_shards": "$body._shards", "hits": "$body.hits",/]
+// TESTRESPONSE[s/"buckets": \[\]/"buckets":\[\{"key":1070000.0,"doc_count":5\}\]/]
+
+<1> The aggregation type, `histogram`, followed by a `#` separator and the aggregation's name, `my-agg-name`.
+
+[discrete]
+[[use-scripts-in-an-agg]]
+=== Use scripts in an aggregation
+
+Some aggregations support <<modules-scripting,scripts>>. You can
+use a `script` to extract or generate values for the aggregation:
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "histogram": {
+        "interval": 1000,
+        "script": {
+          "source": "doc['my-field'].value.length()"
+        }
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.request.method/]
+
+If you also specify a `field`, the `script` modifies the field values used in
+the aggregation. The following aggregation uses a script to modify `my-field`
+values:
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "histogram": {
+        "field": "my-field",
+        "interval": 1000,
+        "script": "_value / 1000"
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.response.bytes/]
+
+Some aggregations only work on specific data types. Use the `value_type`
+parameter to specify a data type for a script-generated value or an unmapped
+field. `value_type` accepts the following values:
+
+* `boolean`
+* `date`
+* `double`, used for all floating-point numbers
+* `long`, used for all integers
+* `ip`
+* `string`
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "histogram": {
+        "field": "my-field",
+        "interval": 1000,
+        "script": "_value / 1000",
+        "value_type": "long"
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.response.bytes/]
+
+[discrete]
+[[agg-caches]]
+=== Aggregation caches
+
+For faster responses, {es} caches the results of frequently run aggregations in
+the <<shard-request-cache,shard request cache>>. To get cached results, use the
+same <<shard-and-node-preference,`preference` string>> for each search. {es}
+routes searches with the same preference string to the same shards. If the
+shards' data doesn’t change between searches, the shards return cached
+aggregation results.
+
+[discrete]
+[[limits-for-long-values]]
+=== Limits for `long` values
+
+When running aggregations, {es} uses <<number,`double`>> values to hold and
+represent numeric data. As a result, aggregations on <<number,`long`>> numbers
+greater than +2^53^+ are approximate.
```

Comment:
some results are approximate anyway by their very nature. Like cardinality. But I don't have a good way to say that.

Reply:
> some results are approximate anyway by their very nature

That feels more relevant for the docs of an individual agg. This seems more like a limitation that could apply to any agg dealing with numeric data.